src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs

```diff
@@ -93,6 +93,24 @@ instance Arithmetic a => BinaryExpansion (ArithmeticCircuit a) where
     binaryExpansion r = circuits $ runCircuit r >>= expansion (numberOfBits @a)
     fromBinary bits = circuit $ Haskell.traverse runCircuit bits >>= horner
 
+instance Arithmetic a => DiscreteField' (ArithmeticCircuit a) where
+    equal r1 r2 = isZeroC (r1 - r2)
+
+instance Arithmetic a => Trichotomy (ArithmeticCircuit a) where
+    trichotomy r1 r2 =
+        let
+            bits1 = binaryExpansion r1
+            bits2 = binaryExpansion r2
```

Comment:
Are these least significant bit first or most significant bit first? I did the lexicographical fold assuming most significant bit first (I think).